benches/benches/micro_ops.rs

```diff
@@ -0,0 +1,244 @@
+use criterion::{black_box, criterion_group, criterion_main, BatchSize, Criterion};
+use hydroflow::hydroflow_syntax;
+use rand::distributions::Distribution;
+use rand::distributions::Uniform;
+use rand::rngs::StdRng;
+use rand::SeedableRng;
+
+fn ops(c: &mut Criterion) {
+    let mut rng = StdRng::from_entropy();
+
+    c.bench_function("micro/ops/identity", |b| {
+        b.iter_batched_ref(
+            || {
+                const NUM_INTS: usize = 10_000;
+                let dist = Uniform::new(0, 100);
+                let data: Vec<usize> = (0..NUM_INTS).map(|_| dist.sample(&mut rng)).collect();
+
+                hydroflow_syntax! {
+                    source_iter(black_box(data)) -> identity() -> for_each(|x| { black_box(x); });
+                }
+            },
+            |df| {
+                df.run_available();
```

Comment:
`source_iter` only emits its values once, seems like that leads to all the later `df.run_available()`s to do nothing?

Reply:
each iteration should get called with a freshly created hydroflow_syntax!{}, if I am understanding how iter_batched is working.

Reply:
this is the source:

```rust
                let mut input = black_box(setup());

                let start = self.measurement.start();
                let output = routine(&mut input);
                let end = self.measurement.end(start);
                self.value = self.measurement.add(&self.value, &end);
                ```

Reply:
Ok, that looks right then

Reply:
Is there no way to get the macro-expansion to happen once and then reuse the object for multiple runs?

Reply:
> Is there no way to get the macro-expansion to happen once and then reuse the object for multiple runs?

This is possible, but it would require using a channel to feed new data into the object to re-use it for each iteration. There is some amount of overhead associated with pulling out of a channel when compared to using a source_iter I would think. The construction of the object using batched_iter is not timed.